$OpenBSD: patch-deps_eleveldb_c_src_detail_hpp,v 1.1 2013/03/26 05:26:30 jmatthew Exp $

define inc_and_fetch and dec_and_fetch for size_t
https://github.com/basho/eleveldb/pull/54

--- deps/eleveldb/c_src/detail.hpp.orig	Wed Jan 30 10:06:04 2013
+++ deps/eleveldb/c_src/detail.hpp	Tue Mar 12 17:13:26 2013
@@ -90,7 +90,7 @@ inline uint32_t inc_and_fetch(volatile uint32_t *ptr)
 #endif
 }
 
-#ifdef __APPLE__
+#if defined(__APPLE__) || defined(__OpenBSD__)
 template <>
 inline size_t inc_and_fetch(volatile size_t *ptr)
 {
@@ -121,7 +121,7 @@ inline uint32_t dec_and_fetch(volatile uint32_t *ptr)
 #endif
 }
 
-#ifdef __APPLE__
+#if defined(__APPLE__) || defined(__OpenBSD__)
 template <>
 inline size_t dec_and_fetch(volatile size_t *ptr)
 {
